### PR TITLE
Issue #3053 Remove Boot2Docker from documentation

### LINKED ISSUE #3053: Remove b2d ISO from documentation
**Note:** This will leave the CentOS ISO as the only supported option. Move the "Choosing the ISO Image" section to the advanced usage section of the documentation accordingly.

### PR DESCRIPTION
**Pull request by robin-owen**
Fixes #3053.

This also removes references to the Minikube ISO.

### CODE REVIEW

File: docs/source/troubleshooting/troubleshooting-misc.adoc [hunk 1]
```diff
@@ -34,33 +34,6 @@ Depending on your operating system and shell environment, certain special charac
...
-
-The reason is an incompatibility between the Docker client shipped with your host and the Docker daemon which is running inside the {project} VM.
-
-Workaround: Use the xref:../using/basic-usage.adoc#choosing-iso-image[CentOS ISO] when using {project} on Fedora, CentOS or RHEL.
-For example:
-
-----
-minishift start --iso-url centos
-----
-
```
Comment (praveenkumar) [inline docs/source/troubleshooting/troubleshooting-misc.adoc]: 👍

**Comment by LalatenduMohanty** [comment]
Building the doc here https://ci.centos.org/job/minishift-docs/62/ to cross check the formatting

**Comment by LalatenduMohanty** [comment]
@robin-owen Have you built this locally and checked if the formatting is fine?

**Comment by robin-owen** [comment]
@LalatenduMohanty Apologies for the late reply -- yes, I have built this locally and verified that the formatting is correct and all of the pages link together appropriately.

**Pull request merged**